src/main/java/cdw/springtraining/gatekeeper/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,47 @@
+package cdw.springtraining.gatekeeper.security;
+
+
+import cdw.springtraining.gatekeeper.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import cdw.springtraining.gatekeeper.entites.User;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    UserRepository userRepository;
+    @Autowired
+    public CustomUserDetailsService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+
+        User user = (User) userRepository.findByUserName(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User not exists by Username or Email"));
```

Comment:
try to maintain a constants file for the strings

Reply:
Changed to constatnts